pkg/spoke/hubclientcert/certificate.go

```
@@ -0,0 +1,215 @@
+//   2. TLSKeyFile exists
+//   3. TLSCertFile exists and the certificate is not expired
+func hasValidKubeconfig(secret *corev1.Secret) bool {
+	if secret.Data == nil {
+		klog.V(4).Infof("No kubeconfig found in secret %q", secret.Namespace+"/"+secret.Name)
+		return false
+	}
+
+	if _, ok := secret.Data[KubeconfigFile]; !ok {
+		klog.V(4).Infof("No %q found in secret %q", KubeconfigFile, secret.Namespace+"/"+secret.Name)
+		return false
+	}
+
+	if _, ok := secret.Data[TLSKeyFile]; !ok {
+		klog.V(4).Infof("No %q found in secret %q", TLSKeyFile, secret.Namespace+"/"+secret.Name)
+		return false
+	}
+
+	certData, ok := secret.Data[TLSCertFile]
+	if !ok {
+		klog.V(4).Infof("No %q found in secret %q", TLSCertFile, secret.Namespace+"/"+secret.Name)
+		return false
+	}
+
+	valid, err := IsCertificatetValid(certData)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("unable to validate certificate in secret %s", secret.Namespace+"/"+secret.Name))
+		return false
+	}
+
+	return valid
+}
+
+// LoadClientConfig loads client config from a kubeconfig file
+func LoadClientConfig(kubeconfigPath string) (*restclient.Config, error) {
+	// clientcmd.RESTConfigFromKubeConfig() and clientcmd.NewClientConfigFromBytes()
+	// do not handle very well when the kubeconfig file references cert/key files
+	// with relative file paths
+	loader := &clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfigPath}
+	loadedConfig, err := loader.Load()
+	if err != nil {
+		return nil, err
+	}
+
+	return clientcmd.NewNonInteractiveClientConfig(
+		*loadedConfig,
+		loadedConfig.CurrentContext,
+		&clientcmd.ConfigOverrides{},
+		loader,
+	).ClientConfig()
+}
+
+// IsCertificatetValid return true if all certs in client certificate are not expired.
+// Otherwise return false
+func IsCertificatetValid(certData []byte) (bool, error) {
+	certs, err := certutil.ParseCertsPEM(certData)
+	if err != nil {
+		return false, errors.New("unable to parse certificate")
+	}
+
+	if len(certs) == 0 {
```

Please return an error in this case.